lib/std/net.zig

```diff
@@ -179,7 +179,7 @@ pub const Address = extern union {
                     unreachable;
                 }
 
-                const path_len = std.mem.len(std.meta.assumeSentinel(&self.un.path, 0));
+                const path_len = std.mem.sliceTo(&self.un.path, 0).len;
```

Comment:
This change is a bug fix and a great example of why restricting `mem.span()` and `mem.len()` to sentinel terminated pointers is necessary. This code worked correctly before it was subtly broken in 48660371a2f66b3859831abb276180c557a12f93, which changed the type passed to `mem.span()` from a sentinel terminated pointer to a sentinel terminated array and thus cause `mem.span()` to stop iterating to find the sentinel but rather return the full length of the `self.un.path` array